packages/clickable/src/use-event-listeners.ts

```diff
@@ -42,11 +43,11 @@ export function useEventListeners(): EventListeners {
 
   React.useEffect(() => {
     return () => {
-      listeners.current.forEach((value, key) => {
+      currentListeners.forEach((value, key) => {
         remove(value.el, value.type, key, value.options)
       })
     }
-  }, [remove])
+  }, [remove, currentListeners])
```

Comment:
I think we could safely remove `currentListener` from the deps array, if we use `listeners.current` inside useEffect.

Reply:
This threw an error case of `exhaustive-deps` that I hadn't seen before:

```
The ref value 'listeners.current' will likely have changed by the time this effect cleanup function runs. If this ref points to a node rendered by React, copy 'listeners.current' to a variable inside the effect, and use that variable in the cleanup function.eslintreact-hooks/exhaustive-deps
```

Reply:
Good to know 🤔 👍

Reply:
Though I guess since it specifically says "If this ref points to a node rendered by React" and it's not a node rendered by React, we could probably ignore this case?

Reply:
If the event listeners are updated and the effect runs again, removing all listeners, we have no listeners left. This is a state we do not want to have, do we? We would have to call add again to reregister them, because they are not reregistered in the same effect.

Reply:
I think we should go back to the prev version and ignore the linter warning.

This is what happens, if I play it through:

1. first render - `listeners.current` is an empty `Map`
2. something calls `add()` -> `listeners.current` has 1 entry
3. cleanup effect runs, because `currentListeners` has changed -> `listeners.current` is empty -> not good

Reply:
I think I'm gonna test this to make sure. I feel like #3 is what the rule is helping us to avoid, where if you unmount immediately after 2, `listeners.current` wouldn't have updated inside the cleanup callback yet to include that new entry before the cleanup is called, so you're left with an orphan listener that wasn't removed, but setting `currentListeners` in the effect body and using it in cleanup avoids that.

Reply:
I thought about this a few times in the past day 😅 
The reference in `currentListeners` or `listeners.current` will never change, because the `Map` never gets recreated when adding or removing an entry. 
So the new and the previous version both work as expected.
IMO we can leave it as it is and follow the linter rule. 

What a ride :D

Reply:
Round and round we go 😆